### PR TITLE
aggregate distinct customer ids in email rollup

### PR DESCRIPTION
**Pull request by fivetran-jamie**
**Are you a current Fivetran customer?** 

internal

**What change(s) does this PR introduce?** 

- adds a `distinct` to the `fivetran_utils.string_agg` of customer ids in `int_shopify__customer_email_rollup`
- adds extra line to README about the shopify holistic reporting package having a dependency on this package
- adds `lower()` to some spots we are looking at transaction.kind (i have always seen it as lowercase in the source data but just in case -- i changed this while investigating something and felt it didn't hurt to leave in) 
- remove unused fields from `int_shopify__customer_email_rollup`

**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [x] No  (please provide an explanation as to how the change is non-breaking below.)

no columns were changed -- the `customer_ids` field will just not aggregate duplicate IDs 

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [x] No but came out of validating the solution for https://github.com/fivetran/dbt_shopify_holistic_reporting/issues/14  and https://github.com/fivetran/dbt_shopify_holistic_reporting/issues/12

**How did you test the PR changes?** 

 

- [x] Buildkite  
- [x] Local (please provide additional testing details below)

before change: 
image

after change:
image

Aggregated metrics tie out as well 

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [ ] Redshift
- [ ] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
👾 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

**Review by fivetran-avinash — CHANGES_REQUESTED** [review]
Hey @fivetran-jamie! So it looks like the customer ids are distinct now, but the aggregation is still not changing and doesn't match with the order counts and amount summations for customer ids with multiple tags. Can you take a look at the issues I saw in `int_shopify__customer_email_rollup` and see if there's validity to what I'm seeing? Thanks!

File: models/intermediate/int_shopify__customer_email_rollup.sql [hunk 1]
```diff
@@ -21,7 +21,7 @@ with customers as (
         customers.source_relation,
 
         -- fields to string agg together
-        {{ fivetran_utils.string_agg("cast(customers.customer_id as " ~ dbt.type_string() ~ ")", "', '") }} as customer_ids,
+        {{ fivetran_utils.string_agg("distinct cast(customers.customer_id as " ~ dbt.type_string() ~ ")", "', '") }} as customer_ids,
```
Comment (fivetran-avinash) [inline models/intermediate/int_shopify__customer_email_rollup.sql:24]: See Height regarding potential concerns of duplicated aggregates.

Comment (fivetran-avinash) [reply models/intermediate/int_shopify__customer_email_rollup.sql:24]: This is resolved now.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+
+## Bug Fixes
+- Adjusts the `int_shopify__customer_email_rollup` model to aggregate _distinct_ `customer_ids`.
+- Ensures that each order is tagged with the orderer's `email` before aggregating order metrics in `int_shopify__emails__order_aggregates`.
+
+## Under the Hood
+- Ensures transaction `kinds` are being read correctly by applying a `lower()` function. 
+
+## Related Releases:
+- https://github.com/fivetran/dbt_shopify_holistic_reporting/releases/tag/v0.4.0
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply CHANGELOG.md:14]: i changed the title to "Related-Package Releases" 🤷

[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply CHANGELOG.md:14]: Works for me!

**Review by fivetran-avinash — APPROVED** [review]
All's good! Whenever Databricks stops timing out you're good to release.

**Pull request merged**